core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -240,23 +256,46 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
     this.languageDropdownShown = !this.languageDropdownShown;
   }
 
-  toggleActivityDropdown(): void {
-    this.activityDropdownShown = !this.activityDropdownShown;
+  getDateThatIsDaysBeforeToday(numberOfDaysBeforeToday: number): Date {
+    const today = new Date();
+    return new Date(
+      today.getTime() - this.ONE_DAY_IN_MILLIS * numberOfDaysBeforeToday
+    );
+  }
+
+  getNumberOfDaysForDateBeforeToday(date: Date): number {
```

Comment:
Could you please explain what this function does and why it needs?

Reply:
This function returns number of days before today corresponding to the give given date. For example, today is 23 may 2024, so for date = 16 may 2024 this function returns 7 .